lib/api/CDataFrameAnalyzer.cc

```diff
@@ -126,19 +126,20 @@ void CDataFrameAnalyzer::run() {
 
     LOG_TRACE(<< "Running analysis...");
 
-    CDataFrameAnalysisRunner* analysis{m_AnalysisSpecification->run(*m_DataFrame)};
-
-    if (analysis == nullptr) {
-        return;
-    }
-
     // We create the writer in run so that when it is finished destructors
     // get called and the wrapped stream does its job to close the array.
 
     // TODO Revisit this can probably be core::CRapidJsonLineWriter.
     auto outStream = m_ResultsStreamSupplier();
     core::CRapidJsonConcurrentLineWriter outputWriter{*outStream};
 
+    CDataFrameAnalysisRunner* analysis{m_AnalysisSpecification->run(*m_DataFrame)};
+    if (analysis == nullptr) {
+        return;
+    }
```

Comment:
Can't we move this back to where it was before, i.e. no need to create the stream if it isn't going to be used.

Reply:
Code changed as discussed.